relentless/volume.py

```diff
@@ -49,29 +57,72 @@
     :members:
 .. autoclass:: Cube
     :members:
+.. autoclass:: Parallelogram
+    :members:
 
 """
 import abc
 from enum import Enum
 
 import numpy
 
-class Volume(abc.ABC):
+class Extent(abc.ABC):
     r"""Abstract base class defining a region of space.
 
-    A Volume can be any region of space; typically, it is a simulation "box."
+    An Extent can be any region of space; typically, it is a simulation "box."
     Any deriving class must implement the volume method that computes the scalar
     volume of the region. Additionally, methods to serialize and deserialize a
     Volume must be specified so that the object can be saved to disk.
 
     """
     @property
+    def extent(self):
+        r"""float: Extent of the region."""
+        pass
+    
+    @classmethod
     @abc.abstractmethod
-    def volume(self):
-        r"""float: Volume of the region."""
+    def to_json(self):
+        r"""Serialize as a dictionary.
+
+        The serialized data can be saved to file as JSON data.
+
+        Returns
+        -------
+        dict
+            The serialized :class:`Volume` data.
+
+        """
         pass
 
+    @classmethod
     @abc.abstractmethod
+    def from_json(cls, data):
+        r"""Deserialize from a dictionary.
+
+        Returns
+        -------
+        :class:`Volume`
+            The object reconstructed from the ``data``.
+
+        """
+        pass
+
+class Volume(Extent):
+    r"""Derived class defining a 3d region of space.
+
+    A Volume can be any region of space; typically, it is a simulation "box."
+    Any deriving class must implement the volume method that computes the scalar
+    volume of the region. Additionally, methods to serialize and deserialize a
+    Volume must be specified so that the object can be saved to disk.
+
+    """
+    @property
+    def volume(self):
+        r"""float: Volume of the region."""
+        return self.extent
+
+    @classmethod
```

Comment:
You shouldn't implement `to_json` or `from_json` methods here because `Volume` is essentially a passthrough to other deriving classes. This will leave them as `abstractmethod`s so that an error is raised if they don't get implemented.